jobs/validate/nagios-spec.yml

```diff
@@ -0,0 +1,82 @@
+plan:
+  - &BASE_JOB
+    env:
+      - SNAP_VERSION=1.17/edge
+      - JUJU_DEPLOY_BUNDLE=cs:~containers/charmed-kubernetes
+      - JUJU_DEPLOY_CHANNEL=edge
+      - JUJU_CLOUD=aws/us-east-2
+      - JUJU_CONTROLLER=validate-ck-nagios
+      - JUJU_MODEL=validate-nagios
+    before-script:
+      - juju:
+          cloud: $JUJU_CLOUD
+          controller: $JUJU_CONTROLLER
+          model: $JUJU_MODEL
+          bootstrap:
+            constraints: "arch=amd64"
+            debug: no
+            replace-controller: yes
+            model-default:
+              - test-mode=true
+              - resource-tags=owner=k8sci
+          deploy:
+            reuse: no
+            bundle: $JUJU_DEPLOY_BUNDLE
+            overlay: |
+              applications:
+                kubernetes-master:
+                  options:
+                    channel: $SNAP_VERSION
+                kubernetes-worker:
+                  options:
+                    channel: $SNAP_VERSION
+            wait: yes
+            timeout: 7200
+            channel: $JUJU_DEPLOY_CHANNEL
```

Comment:
You can re-use 'spec-helpers' like https://github.com/charmed-kubernetes/jenkins/blob/master/jobs/validate/spec.yml#L10-L12